### PR TITLE
refactor: make signup modal copies dynamic

### PR DESCRIPTION
**Pull request by ayadotdev**
Added AB testing for signup modal title and buttons copies. 
Added AB testing to hide a profile image.

DD-416 #done

### CODE REVIEW

File: packages/webapp/pages/register.tsx
Comment (rebelchris) [inline packages/webapp/pages/register.tsx:62]: For this one it would be better to use `isFeaturedEnabled` 

It worked like this:

```js
{!isFeaturedEnabled(Features.HideSignupModalProfileImage) && <EditImageWithJoinedDate user={user} />}
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
One small thing, other than that, it looks amazing 👏
Well done 🎉

**Comment by ayadotdev** [comment]
Note:
Removing a profile image there is a space between headings and the Profile form which has some margin.

Screenshot 2022-01-06 at 18 59 13

**Review by sshanzel — CHANGES_REQUESTED** [review]
It is super awesome that you noted the empty space when the profile image is hidden, the link below to the SC should be able to help with how much the margin must be applied - the spacing should exactly be `24px`.

Let me know if you need help navigating through SC 🚢 

https://www.sketch.com/s/28f2d32c-2923-45b9-91d5-ad7c27be17c9/a/7yr21AP#Inspector

File: packages/webapp/pages/register.tsx
Comment (rebelchris) [inline packages/webapp/pages/register.tsx:35]: Oh sorry my bad, I see we don't have this setup in the `featureManagement` here.
This should be sitting in there as soon as we merge our current branch.

For now you could add the following code to the `featureManagement` file so we have it aligned.

```js

export const isFeaturedEnabled = (key: Features, flags: IFlags): boolean =>
  flags[key?.id]?.enabled;
```

See the following PR for the new file:
https://github.com/dailydotdev/apps/blob/DD-354-my-feed/packages/shared/src/lib/featureManagement.ts

File: packages/shared/src/lib/featureManagement.ts
Comment (rebelchris) [inline packages/shared/src/lib/featureManagement.ts:38]: When using the boolean checked values, we can remove the default value here.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Almost there!

My bad for forgetting we don't have this in master yet.

**Review by rebelchris — APPROVED** [review]
Nice this is perfect, if you can just check the margin note from Lee it will be done 🎉

**Review by sshanzel — APPROVED** [review]
LGTM! 🚀

File: packages/webapp/pages/register.tsx
Comment (idoshamun) [inline packages/webapp/pages/register.tsx:32]: Not sure if we need this function actually...

**Review by idoshamun — COMMENTED** [review]
Looks great I had just one optimization comment but nothing major.
Anyway, for next time it's better to wait for my review before merging :)

File: packages/webapp/pages/register.tsx
Comment (ayadotdev) [reply packages/webapp/pages/register.tsx:32]: @idoshamun I added that function for readability purpose. But you are right, it may be redundant. Happy to remove it if you feel it is better without :thumbsup:

Comment (idoshamun) [reply packages/webapp/pages/register.tsx:32]: Let it be for now 😁

**Pull request merged**
